plugins/outputs/influxdb_v2/http.go

```diff
@@ -315,13 +341,27 @@ func (c *httpClient) writeBatch(ctx context.Context, b *batch) error {
 	}
 
 	// Execute the request
+	start := time.Now()
 	resp, err := c.client.Do(req.WithContext(ctx))
+	durationMs := time.Since(start).Milliseconds()
 	if err != nil {
+		c.statFailedWriteDuration.Set(durationMs)
+		c.statFailedWrites.Incr(1)
+		var urlErr *url.Error
+		if errors.As(err, &urlErr) && urlErr.Timeout() {
+			c.statTimeout.Incr(1)
+			c.statRetryableErrorCounters.Incr(1)
+		} else {
+			c.statNonRetryableErrorCounters.Incr(1)
+		}
```

Comment:
I think this is misleading. If a request is retryable or not, not only depends on if it is a timeout error (see the status-code reasoning below)... I guess you mean
```suggestion
		if errors.As(err, &urlErr) && urlErr.Timeout() {
			c.statTimeout.Incr(1)
		}
```